src/lib.rs

```diff
@@ -997,9 +1130,23 @@ mod test {
 	use bitcoin_hashes::hex::FromHex;
 	use bitcoin_hashes::sha256::Sha256Hash;
 
+	#[test]
+	fn test_system_time_bounds_assumptions() {
+		use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+		/// The upper and lower bounds of `SystemTime` are not part of its public contract and are
+		/// platform specific. That's why we have to test if our assumptions regarding these bounds
+		/// hold on the target platform.
+		///
+		/// If this test fails on your platform, please don't use the library and open an issue
+		/// instead so we can resolve the situation. Currently this library is tested on:
+		///   * Linux (64bit)
+		let _ = UNIX_EPOCH + Duration::from_secs(::std::i64::MAX as u64);
```

Comment:
Is this guaranteed to panic on all platforms? Maybe (UNIX_EPOCH - res).as_secs() on the result and double-check that you get back i64::MAX?

Reply:
I think it is guaranteed to panic in debug mode if an overflow occurs.

I made the test use the `SYSTEM_TIME_MAX_UNIX_TIMESTAMP` constant instead of `i64::MAX` (since that's our actual assumption) and added some more tests.